### PR TITLE
Box: Remove deprecated color options

### PR DESCRIPTION
**Pull request by EduhCosta**
#### What changed?

Remove all deprecated color tokens, following the table reference;
Conversion plan: https://coda.io/d/Gestalt-Team-Folder_d2LeXkQ1kVX/Box-Color-Deprecation-Plan_suv2k#_luUsa

The following codemod help you to do the follow table replacement.

```bash
yarn codemod modifyPropValue ~/path/to/your/code
 --component=Box
 --previousProp=color
 --previousValue=<color>
 --nextValue=<new color>
 ```
To detect manual changes use

```bash
 yarn codemod detectManualReplacement ~/path/to/your/code
 --component=Box
 --prop=color
 --value=<color>
 ```

Screenshot 2022-07-07 at 19 15 18

#### Why?

Improve the consistence and semantic names and usages of each color on Box component.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4163)

### CODE REVIEW

File: packages/gestalt/src/Box.js [hunk 1]
```diff
@@ -136,27 +136,10 @@ type Props = {
    * Default: 'transparent'
    */
   color?:
```
Comment (ayeshakmaz) [inline packages/gestalt/src/Box.js:138]: I believe the boxTransforms file also needs to be updated

Comment (EduhCosta) [reply packages/gestalt/src/Box.js:138]: Done

Comment (ayeshakmaz) [reply packages/gestalt/src/Box.js:138]: You'll also need to remove all current instances of the old colors from our docs examples, particularly in Box docs. See slack for details

**Comment by AlbertCarreras** [comment]
For the tests failing try to figure out https://gestalt.pinterest.systems/development#Create-a-pull-request if not let's chat nd see what's going on

**Review by AlbertCarreras — APPROVED** [review]
LGTM, let's merge after @ayeshakmaz R+

**Review by ayeshakmaz — APPROVED** [review]

**Pull request merged**